File: multi_lang/node_templates/test_template.js
// Template: Jest tests (raw). Placeholders substituted by NodeTestGenerator.
const request = require('supertest');
const app = require('./__MODULE_NAME__');

describe('__MODULE_NAME__ API', () => {
  test('GET /healthz returns ok', async () => {
    const res = await request(app).get('/healthz');
    expect(res.statusCode).toBe(200);
    expect(res.body.status).toBe('ok');
  });

  test('GET /api/v1/__ROUTE__ returns listing', async () => {
    const res = await request(app).get('/api/v1/__ROUTE__');
    expect(res.statusCode).toBe(200);
  });
});

File: multi_lang/python_templates/test_template.py
"""Template: pytest unit tests (raw). Placeholders are substituted by PythonTestGenerator."""
import pytest
from fastapi.testclient import TestClient

from __MODULE_NAME__ import __MODULE_NAME__app as app


@pytest.fixture
def client():
    """Provide a FastAPI test client for the generated service."""
    with TestClient(app) as test_client:
        yield test_client


def test_healthz(client) -> None:
    """Verify the liveness probe returns 200 OK."""
    response = client.get("/healthz")
    assert response.status_code == 200
    assert response.json()["status"] == "ok"


def test_list_items(client) -> None:
    """Verify the __MODULE_NAME__ listing endpoint responds."""
    response = client.get("/api/v1/__ROUTE__")
    assert response.status_code == 200